### ISSUE TITLE
Increased calls to exwm-input--on-buffer-list-update

### ISSUE THREAD

**Issue by adrianparvino** [ch11ng/exwm#648]
Between 0.22 and master, the calling of "exwm-input--on-buffer-list-update" has increased. I'm on Spacemacs, which might have contributed to this.

**Comment by ch11ng** [comment]
Did you notice this from debug log? We did loosen the restrictions in a1cf0d9b85e1d1d4f3b50a9d51980688b7b816df.

**Comment by adrianparvino** [comment]
Yep! That might actually be the reason. It happens when moving between workspaces, which almost always have \*spacemacs\* as a target. I can't test it right now since I rolled back to 22 and cherry picked 5b9f4b0.

**Comment by ch11ng** [comment]
Does it cause any problem?

**Comment by adrianparvino** [comment]
Yep. :/ Upon switching to another workspace, it triggers it to switch back to the old window(iff it was an X window), or even worse, it sometimes occurs on \*buffers\*, such as \*XELB-DEBUG\*, pushing you into an X window.

**Comment by ch11ng** [comment]
How do you switch to other workspaces? Have you tried other ways (`exwm-workspace-switch`, `exwm-workspace-switch-create`, or even from the menu)?

**Comment by adrianparvino** [comment]
I've only tried with `exwm-workspace-switch`. I assume "from the menu" is from the modeline?

**Comment by ch11ng** [comment]
Yes, the 'EXWM' menu on the mode-line.  Did you invoke `exwm-workspace-switch` interactively?
